src/internal/m365/collection/drive/url_cache.go

```diff
@@ -250,10 +251,10 @@ func (uc *urlCache) updateCache(
 			}
 		}
 
-		itemID := ptr.Val(item.GetId())
+		itemID := strings.Clone(ptr.Val(item.GetId()))
```

Comment:
can you please leave a comment for why this is necessary?  That'll alleviate confusion for the next maintainer.